iib/workers/tasks/build_create_empty_index.py

```diff
@@ -0,0 +1,158 @@
+
+log = logging.getLogger(__name__)
+
+
+def _get_present_operators(from_index, base_dir):
+    """
+    Get a list of operators already present in the index image.
+
+    :param str from_index: index image to inspect.
+    :param str base_dir: base directory to create temporary files in.
+    :return: list of unique present operators as provided by the grpc query
+    :rtype: list, list
+    :raises IIBError: if any of the commands fail.
+    """
+    db_path = _get_index_database(from_index, base_dir)
+    port, rpc_proc = _serve_index_registry(db_path)
+
+    # doc: operators https://www.redhat.com/ja/blog/deploying-operators-olm-bundles
+
+    operators = run_cmd(
+        ['grpcurl', '-plaintext', f'localhost:{port}', 'api.Registry/ListPackages'],
+        exc_msg='Failed to get operator data from index image',
+    )
+    rpc_proc.kill()
+
+    # If no data is returned there are not operators present
+    if not operators:
+        return []
+
+    # Transform returned data to parsable json
+    present_operators = []
+    new_operators = [json.loads(operator) for operator in re.split(r'(?<=})\n(?={)', operators)]
+
+    for operator in new_operators:
+        present_operators.append(operator['name'])
+
+    return present_operators
+
+
+@app.task
+@request_logger
+def handle_create_empty_index_request(
+    from_index, request_id, binary_image=None, labels=None, binary_image_config=None
+):
+    """Coordinate the the work needed to create the index image with labels.
+
+    :param str from_index: the pull specification of the container image containing the index that
+        the index image build will be based from.
+    :param int request_id: the ID of the IIB build request
+    :param str binary_image: the pull specification of the container image where the opm binary
+        gets copied from.
+    :param dict labels: the dict of labels required to be added to a new index image
+    :param dict binary_image_config: the dict of config required to identify the appropriate
+        ``binary_image`` to use.
+    """
+    _cleanup()
+
+    # which distribution scope is the correct one?
+    # distribution_scope or com.redhat.index.delivery.distribution_scope
+    dist_scope = labels.get('com.redhat.index.delivery.distribution_scope', None)
+
+    # overwrite_from_index_token is needed
+    prebuild_info = prepare_request_for_build(
+        request_id,
+        RequestConfigCreateIndexImage(
+            _binary_image=binary_image,
+            from_index=from_index,
+            distribution_scope=dist_scope,
+            binary_image_config=binary_image_config,
+        ),
+    )
+    from_index_resolved = prebuild_info['from_index_resolved']
+
+    prebuild_info['labels'] = labels
+
+    _update_index_image_build_state(request_id, prebuild_info)
+    with tempfile.TemporaryDirectory(prefix='iib-') as temp_dir:
+        if from_index:
+            msg = 'Checking operators present in index image'
+            log.info(msg)
```

Comment:
Just question. This log is here for some purpose. Perhaps to have better and cleaner log file?
I am just pointing this out because `set_request_state` logs change of state.

Reply:
this was for testing purposes.Removing